### PR TITLE
feat(flutter): Add auth next steps

### PR DESCRIPTION
**Pull request by dnys1**
#### Description of changes:

Follows the existing [iOS](https://docs.amplify.aws/lib/auth/signin_next_steps/q/platform/ios/) and [Android](https://docs.amplify.aws/lib/auth/signin_next_steps/q/platform/android/) docs for the next steps encountered during a sign-in flow.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [x] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

File: src/fragments/lib/auth/flutter/signin_next_steps/10_signin.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+The `Amplify.Auth.signIn` API returns a `SignInResult` object. To see if additional signin steps are required, Inspect the `nextStep.signInStep` property of the result.
+When `result.isSignedIn` is `true` or when the `signInStep` is `done`, the sign in flow is complete.
```
Comment (Equartey) [inline src/fragments/lib/auth/flutter/signin_next_steps/10_signin.mdx:3]: I find this some what confusing. What is `result`? Where does `signInStep` live?

Could we denote the relationship relative to`SignInResult`?

File: src/fragments/lib/auth/flutter/signin_next_steps/20_confirm_sms_mfa.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+If the next step is `confirmSignInWithSmsMfaCode`, Amplify Auth has sent the user a random code over SMS and is waiting for the user to verify that code. 
+To handle this step, your app's UI must prompt the user to enter the code. After the user enters the code, pass the value to the `confirmSignIn` API.
+
+<Callout>
+
+The result includes an `AuthCodeDeliveryDetails` member. It includes additional information about the code delivery, such as the partial phone number of 
+the SMS recipient, which can be used to prompt the user on where to look for the code.
```
Comment (Equartey) [inline src/fragments/lib/auth/flutter/signin_next_steps/20_confirm_sms_mfa.mdx:7]:
```suggestion
the SMS recipient, which can prompt the user on where to look for the code.
```

File: src/fragments/lib/auth/flutter/signin_next_steps/30_confirm_custom_challenge.mdx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    case AuthSignInStep.confirmSignInWithCustomChallenge:
+      final parameters = result.nextStep.additionalInfo;
+      final hint = parameters['hint']!;
+      safePrint(hint); // "Enter the secret code"
+    // ...
+  }
+}
+```
+
+To complete this step, you should prompt the user for the custom challenge answer, and pass the answer to the `confirmSignIn` API.
```
Comment (Equartey) [inline src/fragments/lib/auth/flutter/signin_next_steps/30_confirm_custom_challenge.mdx:19]:
```suggestion
To complete this step, prompt the user for the custom challenge answer, and pass the answer to the `confirmSignIn` API.
```

File: src/fragments/lib/auth/flutter/signin_next_steps/30_confirm_custom_challenge.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  }
+}
+```
+
+<Callout warning>
+
+<b>Special Handling on confirmSignIn</b>
+
+During a confirmSignIn call if `failAuthentication=true` is returned by the Lambda, the session of the request gets invalidated by Cognito.
+In this case, a `NotAuthorizedException` is thrown and you are required to restart the sign-in flow by calling `Amplify.Auth.signIn`.
```
Comment (Equartey) [inline src/fragments/lib/auth/flutter/signin_next_steps/30_confirm_custom_challenge.mdx:39]:
```suggestion
If `failAuthentication=true` is returned by the Lambda, Cognito will invalidate the session of the request.
This is represented by a `NotAuthorizedException`, which requires restarting the sign-in flow by calling `Amplify.Auth.signIn`.
```

File: src/fragments/lib/auth/flutter/signin_next_steps/60_confirm_signup.mdx [hunk 5]
```diff
@@ -0,0 +1,53 @@
+If the next step is `resetPassword`, Amplify Auth requires that the user confirm their email or phone number before proceeding.
+Use the `resendSignUpCode` API to send a new sign up code to the registered email or phone number, followed by `confirmSignUp` 
+to complete the sign up.
+
+See the [confirm sign up](/lib/auth/signin/q/platform/flutter/#register-a-user) docs for more information.
+
+<Callout>
+
+The result includes an `AuthCodeDeliveryDetails` member. It includes additional information about the code delivery, such as the partial phone number of 
+the SMS recipient, which can be used to prompt the user on where to look for the code.
```
Comment (Equartey) [inline src/fragments/lib/auth/flutter/signin_next_steps/60_confirm_signup.mdx:10]:
```suggestion
the SMS recipient, which can prompt the user on where to look for the code.
```

**Review by Equartey — COMMENTED** [review]
Looking good! Just left a couple comments to consider

File: src/fragments/lib/auth/flutter/signin_next_steps/20_confirm_sms_mfa.mdx (hunk 2 above)
Comment (dnys1) [reply src/fragments/lib/auth/flutter/signin_next_steps/20_confirm_sms_mfa.mdx:7]: This is not saying the same thing IMO. It requires that the developer take action with the delivery details to prompt the user.

**Review by abdallahshaban557 — COMMENTED** [review]
@dnys1  - can we also add the message around the addition of new enum values that Harsh is going to add to this page to this PR? I have added you to that thread now.

**Review by abdallahshaban557 — APPROVED** [review]
LGTM

**Review by Equartey — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**